pkg/diagnostics/networkpod/pod.go

```diff
@@ -0,0 +1,194 @@
+package network
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	kcontainer "k8s.io/kubernetes/pkg/kubelet/container"
+	kexec "k8s.io/kubernetes/pkg/util/exec"
+
+	osclient "github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/diagnostics/networkpod/util"
+	"github.com/openshift/origin/pkg/diagnostics/types"
+	sdnplugin "github.com/openshift/origin/pkg/sdn/plugin"
+)
+
+const (
+	CheckPodNetworkName = "CheckPodNetwork"
+)
+
+// CheckPodNetwork is a Diagnostic to check communication between pods in the cluster.
+type CheckPodNetwork struct {
+	KubeClient *kclient.Client
+	OSClient   *osclient.Client
+
+	vnidMap map[string]uint32
+	res     types.DiagnosticResult
+}
+
+// Name is part of the Diagnostic interface and just returns name.
+func (d CheckPodNetwork) Name() string {
+	return CheckPodNetworkName
+}
+
+// Description is part of the Diagnostic interface and just returns the diagnostic description.
+func (d CheckPodNetwork) Description() string {
+	return "Check pod to pod communication in the cluster. In case of ovs-subnet network plugin, all pods should be able to communicate with each other and in case of multitenant network plugin, pods in non-global projects should be isolated and pods in global projects should be able to access any pod in the cluster and vice versa."
+}
+
+// CanRun is part of the Diagnostic interface; it determines if the conditions are right to run this diagnostic.
+func (d CheckPodNetwork) CanRun() (bool, error) {
+	if d.KubeClient == nil {
+		return false, errors.New("must have kube client")
+	} else if d.OSClient == nil {
+		return false, errors.New("must have openshift client")
+	}
+	return true, nil
+}
+
+// Check is part of the Diagnostic interface; it runs the actual diagnostic logic
+func (d CheckPodNetwork) Check() types.DiagnosticResult {
+	d.res = types.NewDiagnosticResult(CheckPodNetworkName)
+
+	pluginName, ok, err := util.GetOpenShiftNetworkPlugin(d.OSClient)
+	if err != nil {
+		d.res.Error("DPodNet1001", err, fmt.Sprintf("Checking network plugin failed. Error: %s", err))
+		return d.res
+	}
+	if !ok {
+		d.res.Warn("DPodNet1002", nil, fmt.Sprintf("Skipping pod connectivity test. Reason: Not using openshift network plugin."))
```

Comment:
If you're going to skip the diagnostic, just do this in CanRun() where it will come out as part of the notice. There's really no reason to warn the user here unless you're going to give them some advice about what to do about it.